wcivf/settings/base.py

```diff
@@ -42,8 +43,36 @@ def repo_root(*x):
 # SECURITY WARNING: don't run with debug turned on in production!
 DEBUG = False
 
-# add to param store?
-ALLOWED_HOSTS = ["*"]
+
+def get_ec2_ip():
+    token_req = requests.put(
+        "http://169.254.169.254/latest/api/token",
+        headers={"X-aws-ec2-metadata-token-ttl-seconds": "21600"},
+        timeout=2,
+    )
+    token_req.raise_for_status()
+    token_req.text
+    ip_req = requests.get(
+        "http://169.254.169.254/latest/meta-data/local-ipv4",
+        headers={"X-aws-ec2-metadata-token": token_req.text},
+        timeout=2,
+    )
+    ip_req.raise_for_status()
+    return ip_req.text
+
+
+# Hosts/domain names that are valid for this site; required if DEBUG is False
+# See https://docs.djangoproject.com/en/1.5/ref/settings/#allowed-hosts
+ALLOWED_HOSTS = [
+    "localhost",
+    "127.0.0.1",
+]
+
+if os.environ.get("DC_ENVIRONMENT") and os.environ.get("FQDN"):
+    ALLOWED_HOSTS.append(os.environ.get("FQDN"))
+    ALLOWED_HOSTS.append(get_ec2_ip())
```

Comment:
We can't rely on "`DC_ENVIRONMENT` is set" as a proxy for "on EC2" in this application because it is also set when we run on lambda, and `169.254.169.254` is not contactable from lambda (also `ALLOWED_HOSTS` doesn't matter for scheduled jobs) so I've taken a slightly different approach here.